compiler/.gitignore

```diff
@@ -2,4 +2,4 @@
 dist
 .cabal-sandbox
 cabal.sandbox.config
-.stack-work/
+.stack-work
```

Comment:
Is `.stack-work` a file or a directory? If it's a directory, then the trailing slash is appropriate.